### PR TITLE
Fix/Add separate param for Mobile API path

### LINKED ISSUE #242: ATF cannot find MOBILE_API.xml outside of `components/interfaces`
Since https://github.com/smartdevicelink/sdl_core/pull/3778 ATF is unable to find and parse the MOBILE_API.xml, causing parallel runs to fall back to the XML within the `data` folder and causing normal runs to fail outright.

Some changes are needed to find the MOBILE_API.xml within the rpc_spec submodule.

Perhaps also some changes could be added so that parallel runs print a warning when falling back to an older API XML.

### PR DESCRIPTION
**Pull request by ShobhitAd**
Fixes #242

### Summary
- Added new params `pathToSDLSource` in `modules/configuration/base_config.lua`.
- Split `pathToSDLInterfaces` into two separate params-`pathToSDLMobileInterface` and `pathToSDLHMIInterface`.

### CODE REVIEW

File: modules/atf/util.lua [hunk 1]
```diff
@@ -65,21 +65,30 @@ end
 -- Checks: SDL Core binary, HMI and MObile API files
 -- Stop ATF execution in case any error
 local function check_required_fields()
+  local function check_api_exists(interface_path, api_file)
+    if (not is_file_exists(interface_path..api_file)) and
+       (not is_file_exists(interface_path.."/"..api_file)) then
+      print("ERROR:"..api_file.."file is not accessible at the specified path: "..interface_path)
```
Comment (iCollin) [inline modules/atf/util.lua:71]:
```suggestion
      print("ERROR: "..api_file.." file is not accessible at the specified path: "..interface_path)
```

**Comment by iCollin** [comment]
One more weird thing I am noticing - using specific paths for the mobile and HMI interfaces doesn't seem to apply as I expect in parallel mode.

I setup my base config's values for `config.pathToSDL` and `config.pathToSDLSource` to point at a build version of Core develop, then I specified the values for `config.pathToSDLMobileInterface` and `config.pathToSDLHMIInterface` to point to APIs from version 4.5.

Running one test, the APIs from version 4.5 are applied as I expect, but in parallel mode, the APIs I specified are not applied and I see this output in the parallel output:

```
cp: '/home/developer/sdl/atf/data/MOBILE_API.xml' and 'data/MOBILE_API.xml' are the same file
cp: '/home/developer/sdl/atf/data/HMI_API.xml' and 'data/HMI_API.xml' are the same file
```

**Review by iCollin — APPROVED** [review]

**Comment by GetmanetsIrina** [comment]
@ShobhitAd, [README.md](https://github.com/smartdevicelink/sdl_atf/blob/master/README.md) file should be updated according to current changes.

**Comment by GetmanetsIrina** [comment]
@ShobhitAd, [README.md](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/README.md) should be updated according to the current update.

**Comment by iCollin** [comment]
@GetmanetsIrina the README was updated with this PR. Were you expecting additional README changes? If so what would they be?

**Comment by GetmanetsIrina** [comment]
I mean README.md in [sdl_atf_test_scripts repo ](https://github.com/smartdevicelink/sdl_atf_test_scripts/tree/develop). There i can see the reference to the removed `pathToSDLInterfaces` parameter.

But this ReadMe.md file from `sdl_atf_test_scripts` repo has the not actual information not only about `pathToSDLInterfaces` parameter, but also in general about the config file, so maybe we can plan the update of this file (ReadMe.md from `sdl_atf_test_scripts` repo) in some general way without a link to this one.
I will create an issue in https://github.com/smartdevicelink/sdl_atf_test_scripts.

**Pull request merged**